### PR TITLE
Add comprehensive unit tests for button components

### PR DESCRIPTION
**Pull request by narumikr**
PR-Header

### 🌸 作業概要 🌸

Created unit tests for all 5 button components (BasicButton, HamburgerButton, ScrollTopButton, StrongButton, StylishButton) with 206 test cases total covering rendering, props handling, theme integration, accessibility, and edge cases.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### 💫 追加/変更/修正内容 🍀

- buttonの単体テスト実装

### 🎤 確認事項 🎪

- [x] branch名はどんな作業を行ったかわかる命名にしたか(ex.20250831_add_readme)
- [ ] Pushする前にビルド行い、distを最新にしたか
- [ ] Dev環境にデプロイしての動作確認実施したか
- [ ] アクセシビリティチェックの実施(ave DevToolsの実施)したか
- [ ] 影響確認を行ったか
- [ ] AIによるレビューの実施を行い、以下の確認をしたか
  - [ ] 指摘事項の妥当性の確認
  - [ ] 根拠の確認・調査
  - [ ] 指摘事項の修正
  - [ ] 再レビューの実施

#### 🎸 AI指摘事項 🎹

<Details><Summary>Review by ClaudeAI and ChatGPT</Summary>

```ts
本PRでは指摘事項なし
```

</Details>

### 💻 評価項目 🆚

- 

### 🎵 備考 💚 Y(_ﾟ□ﾟ_) < (あれば)

PR-Foorter

### CODE REVIEW

File: test/components/button/scrollTopButton.test.tsx [hunk 1]
```diff
@@ -0,0 +1,542 @@
...
+
+      await showButton()
+
+      await waitFor(() => {
+        expect(screen.getByTestId('chevron-icon')).toBeInTheDocument()
+      })
+    })
+
+    it('should render with custom id', async () => {
+      const { container } = render(<ScrollTopButton {...defaultProps} id="custom-scroll-button" />)
```
[the remark this answers is not in the document]
Comment (Narumikr) [reply test/components/button/scrollTopButton.test.tsx]: 未使用変数を削除

File: test/components/button/scrollTopButton.test.tsx [hunk 2]
```diff
@@ -0,0 +1,542 @@
...
+      await showButton()
+
+      await waitFor(() => {
+        const button = screen.getByTestId('chevron-icon').closest('button')
+        expect(button).toHaveAttribute('id', 'custom-scroll-button')
+      })
+    })
+
+    it('should render with custom className', async () => {
+      const { container } = render(<ScrollTopButton {...defaultProps} className="custom-class" />)
```
[the remark this answers is not in the document]
Comment (Narumikr) [reply test/components/button/scrollTopButton.test.tsx]: 未使用変数を削除

File: test/components/button/scrollTopButton.test.tsx [hunk 3]
```diff
@@ -0,0 +1,542 @@
...
+
+      await waitFor(() => {
+        const button = screen.getByTestId('chevron-icon').closest('button')
+        expect(button).toHaveClass('custom-class')
+      })
+    })
+
+    it('should render with custom styles', async () => {
+      const customStyle = { backgroundColor: 'red' }
+      const { container } = render(<ScrollTopButton {...defaultProps} style={customStyle} />)
```
[the remark this answers is not in the document]
Comment (Narumikr) [reply test/components/button/scrollTopButton.test.tsx]: 未使用変数を削除
